File: content/problems/2151-maximum-good-people-based-on-statements/solutions/enumerate-good-person-masks/Solution.kt
class Solution {
    fun maximumGood(statements:Array<IntArray>):Int{
        val n=statements.size;
        var answer=0;
        for(mask in 0 until (1 shl n)){
            var valid=true;
            loop@for(i in 0 until n)if(mask and (1 shl i)!=0)for(j in 0 until n)if(statements[i][j]!=2&&statements[i][j]!=(mask shr j and 1)){
                valid=false;
                break@loop
            };
            if(valid)answer=maxOf(answer,Integer.bitCount(mask))
        };
        return answer
    }
}

File: content/problems/2151-maximum-good-people-based-on-statements/solutions/enumerate-good-person-masks/solution.cpp
class Solution {
    public:int maximumGood(vector<vector<int>>& statements){
        int n=statements.size(),answer=0;
        for(int mask=0;mask<(1<<n);++mask){
            bool valid=true;
            for(int i=0;i<n&&valid;++i)if(mask>>i&1)for(int j=0;j<n;++j)if(statements[i][j]!=2&&statements[i][j]!=(mask>>j&1)){
                valid=false;
                break;
            }
            if(valid)answer=max(answer,__builtin_popcount(mask));
        }
        return answer;
    }
};
